src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -261,17 +304,35 @@ public void drive(SwerveModuleState[] states) {
 
 		// Set motor speeds and angles
 		for (int i = 0; i < moduleDriveMotors.length; i++) {
-			if (IS_COMP) {
-				moduleDriveMotors[i].set(
-						states[i].speedMetersPerSecond / MAX_DRIVE_SPEED_METERS_PER_SEC * 12); // set voltage
-			} else {
-				moduleDriveMotors[i].set(
-						states[i].speedMetersPerSecond * DRIVE_VELOCITY_COEFFICIENT); // set velocity
-			}
+			moduleDriveMotors[i].set(
+					states[i].speedMetersPerSecond * DRIVE_VELOCITY_COEFFICIENT); // set velocity
 		}
 		for (int i = 0; i < moduleAngleMotors.length; i++) {
 			moduleAngleMotors[i].set(states[i].angle.getRotations() * STEER_REDUCTION);
 		}
+
+		frontLeftActualVelocityPublisher.set(
+				moduleDriveMotors[0].getVelocity() / DRIVE_VELOCITY_COEFFICIENT);
+		frontRightActualVelocityPublisher.set(
+				moduleDriveMotors[1].getVelocity() / DRIVE_VELOCITY_COEFFICIENT);
+		backLeftActualVelocityPublisher.set(
+				moduleDriveMotors[2].getVelocity() / DRIVE_VELOCITY_COEFFICIENT);
+		backRightActualVelocityPublisher.set(
+				moduleDriveMotors[3].getVelocity() / DRIVE_VELOCITY_COEFFICIENT);
+
+		frontLeftTargetVelocityPublisher.set(
+				states[0].speedMetersPerSecond * DRIVE_VELOCITY_COEFFICIENT);
+		frontRightTargetVelocityPublisher.set(
+				states[1].speedMetersPerSecond * DRIVE_VELOCITY_COEFFICIENT);
+		backLeftTargetVelocityPublisher.set(
+				states[2].speedMetersPerSecond * DRIVE_VELOCITY_COEFFICIENT);
+		backRightTargetVelocityPublisher.set(
+				states[3].speedMetersPerSecond * DRIVE_VELOCITY_COEFFICIENT);
```

Comment:
The target velocity and actual velocity entries will actually have different units, since we are converting motor rotations to meters for the actual velocity but converting meters to motor rotations for the target velocity. To make tuning easier, I'd suggest making them all the same units.